### DISCUSSION TITLE
Issue: Unable to Broadcast Bitcoin Transaction on Network - Error: bad-txns-inputs-missingorspent

### DISCUSSION THREAD

**Discussion by ronaldoguedess** [trustwallet/wallet-core#3581] [Q&A]
### **Issue:**
Unable to broadcast Bitcoin Transaction - bad-txns-inputs-missingorspent

### **Description:**
When attempting to broadcast a Bitcoin transaction, the process fails with the following error: **bad-txns-inputs-missingorspent**.

### **Logs**
```kotlin
2023-11-22 23:08:20.515 E: Planned fee:  9588  amount: 100000  avail_amount: 1214340  change: 1104752
2023-11-22 23:08:20.532 E: Signed BTC transactionx [TESTNET]: 0100000000010193e25259bef539cb58072fcc7f01390b15342a8b219c1ecf78e9853857fd4c9401000000000000000002a08601000000000016001484e93a33d176ca5ca78366c0ac74b414c376325270db100000000000160014280c4bcfe801cff8087fe34ee1c56883aa1c398902473044022002ce5e0fb8e28860ab249ffc755d5992370534559435e891c7e2bbb676bc256f0220211998f9dd14fb53d4ab7cc7be206c915d9dc8ad802c1f5403c7a18804fcfd7101210269d7e7fb9cda349856f6b8a3b23dfa936f196e7e630595f857cfbea2e6333e2800000000
``` 

### **Signed Transaction for Verification on Decode Tool**  
**Transaction Hash:**
0100000000010193e25259bef539cb58072fcc7f01390b15342a8b219c1ecf78e9853857fd4c9401000000000000000002a08601000000000016001484e93a33d176ca5ca78366c0ac74b414c376325270db100000000000160014280c4bcfe801cff8087fe34ee1c56883aa1c398902473044022002ce5e0fb8e28860ab249ffc755d5992370534559435e891c7e2bbb676bc256f0220211998f9dd14fb53d4ab7cc7be206c915d9dc8ad802c1f5403c7a18804fcfd7101210269d7e7fb9cda349856f6b8a3b23dfa936f196e7e630595f857cfbea2e6333e2800000000
[Click here to view on BlockCypher](https://live.blockcypher.com/btc-testnet/decodetx/)

### **To Reproduce:**
After executing the function, success is received along with the signed transaction. However, attempting to broadcast it using [Mempool](https://mempool.space/testnet/tx/push) or [Blockstream](https://blockstream.info/testnet/tx/push) results in the error: {"code":-25,"message":"bad-txns-inputs-missingorspent"}.

### **Kotlin Code:**
```kotlin
private fun signAndBroadcastBitcoinTx(){
  try {
    //TESNET WALLET
    val privateKeyBtc = "---"  
    val fromAddress   = "tb1qzcfl0nmsn3ztf0p8x4nwrlc7zegh7jg5t6507a" //CURRENT WALLET
    val changeAddress = "tb1q9qxyhnlgq88lszrlud8wr3tgsw4pcwvfnrh5yu" //change
    val toAddress     = "tb1qsn5n5v73wm99efurvmq2ca95znphvvjj9mpmn7" //TO
    
    val coinBtc: CoinType = CoinType.BITCOIN
    val scriptMyAddress = BitcoinScript.lockScriptForAddress(fromAddress, coinBtc).data()
    
    // Build a transaction
    val outPoint = Bitcoin.OutPoint.newBuilder()
        .setHash(ByteString.copyFrom("93e25259bef539cb58072fcc7f01390b15342a8b219c1ecf78e9853857fd4c94".toHexBytes()))
        .setIndex(1)
        .build()
    
    val utxo = Bitcoin.UnspentTransaction.newBuilder()
        .setAmount(1214340)
        .setOutPoint(outPoint)
        .setScript(ByteString.copyFrom(scriptMyAddress))
        .build()
    
    val input = Bitcoin.SigningInput.newBuilder()
        .setAmount(100000) // 0.001 BTC in satoshis
        .setHashType(BitcoinSigHashType.ALL.value())
        .setToAddress(toAddress)
        .setChangeAddress(changeAddress)
        .setByteFee(68) // 68 satoshis per byte or 101
        .setCoinType(coinBtc.value())
        .addUtxo(utxo)
        .addPrivateKey(ByteString.copyFrom(privateKeyBtc.hexStringToByteArray())) //.toHexBytes()
    
    // Calculate fee (plan a transaction)
    val plan = AnySigner.plan(input.build(), coinBtc, Bitcoin.TransactionPlan.parser())
    Log(TAG, "Planned fee:  ${plan.fee}  amount: ${plan.amount}  avail_amount: ${plan.availableAmount}  change: ${plan.change}")
    
    // Set the precomputed plan
    input.plan = plan
    input.amount = plan.amount
    
    val output = AnySigner.sign(input.build(), coinBtc, Bitcoin.SigningOutput.parser())
    
    assert(output.error == Common.SigningError.OK)
    val signedTransaction = output.encoded?.toByteArray()
    Log(TAG, "Signed BTC transactionx [TESTNET]: \n${signedTransaction?.toHexString()?.substring(2)}")

  }catch (e: AssertionError){
       Log.e(TAG, "${e.printStackTrace()}")
  }
}
``` 

### **Additional Information:**
-The transaction is successful signed.
-The error occurs during the broadcast attempt using external services.
-The provided logs include the planned fee, amount, available amount, and change.
-The transaction is intended for the Bitcoin TESTNET.
-The Privatekey is for testing purpose.

I appreciate any assistance in resolving this issue. Your support means a lot. 

Thank you!

**Comment by weixuefeng** [comment]
utxo had spend? 
https://mempool.space/testnet/tx/180f6ceb810b51f15c184ad6711ad356b37186a78d0c4c243335e797d4bcdc3f
